src/mysql/result_set.cr

```diff
@@ -72,7 +72,13 @@ class MySql::ResultSet < DB::ResultSet
     if is_nil
       nil
     else
-      @columns[col].column_type.read(row_packet)
+      val = @columns[col].column_type.read(row_packet)
+      # blob 65535, longblob 4294967295, mediumblob 16777215, tinyblob 255
+      if val.is_a?(Slice(UInt8)) && @columns[col].fixedlength != 65535 && @columns[col].fixedlength != 255 && @columns[col].fixedlength != 16777215 && @columns[col].fixedlength != 4294967295
```

Comment:
This won't make it. There will be blobs of that length that should not be interpreted as text.